python_tests/pytest_input_generators.py

```diff
@@ -0,0 +1,312 @@
+# SPDX-FileCopyrightText: Copyright (c) 2023-present NVIDIA CORPORATION & AFFILIATES.
+# All rights reserved.
+# SPDX-License-Identifier: BSD-3-Clause
+# Owner(s): ["module: nvfuser"]
+
+import itertools
+from functools import partial, wraps
+
+import torch
+from torch.testing import make_tensor
+
+from pytest_core import OpInfo, SampleInput, ErrorSample
+from nvfuser import DataType
+
+# TODO Add small value, large value, and extremal-valued samples
+def elementwise_unary_generator(
+    op: OpInfo,
+    dtype: torch.dtype,
+    requires_grad: bool,
+    *,
+    supports_numbers: bool = True,
+    **kwargs,
+):
+    low = None if op.domain.low is None else max(-9, op.domain.low)
+    high = None if op.domain.high is None else min(9, op.domain.high)
+    make_arg = partial(
+        make_tensor,
+        device="cuda",
+        dtype=dtype,
+        low=low,
+        high=high,
+        requires_grad=requires_grad,
+        **kwargs,
+    )
+
+    shapes = (
+        # TODO: restore size zero cases
+        # (0, 2, 1),
+        # (5, 0, 3),
+        (),
+        (11,),
+        (4, 4),
+        (1024, 1024),
+        (64, 64, 64),
+    )
+
+    # Typical inputs
+    for shape in shapes:
+        yield SampleInput(make_arg(shape))
+
+    # Noncontiguous inputs
+    for shape in shapes:
+        yield SampleInput(make_arg(shape, noncontiguous=True))
+
+    # Arbitrarily strided inputs
```

Comment:
Just a note, is overlapping memory something we want to support?

Reply:
Probably could remove these samples.

Reply:
I removed them.